### PR TITLE
Add unit tests and validations for Candidate, User, and Note models

### PR DESCRIPTION
**Pull request by EmilyMB**
- Add tests and validations for models.
- Rename databases with unique names (this solves the issue of the lists and tasks appearing and reappearing in the schema).
- Run **rake db:reset** after this is merged in.

@pascallaliberte @miranmirza @cyprusad

### CODE REVIEW

File: app/models/candidate.rb [hunk 1]
```diff
@@ -1,6 +1,9 @@
 class Candidate < ActiveRecord::Base
   has_many :notes
   has_many :users, through: :notes
+  validates :firstname, presence: true, length: { in: 2..20 }
+  validates :hangout, presence: true, if: "skype.nil?"
+  validates :skype, presence: true, if: "hangout.nil?"
```
Comment (cyprusad) [inline app/models/candidate.rb:6]: I like this!

File: app/models/note.rb [hunk 2]
```diff
@@ -2,6 +2,8 @@ class Note < ActiveRecord::Base
   belongs_to :candidate
   belongs_to :user
   before_save :update_candidate
+  validates :comment, presence: true, length: { in: 10..2000 }
```
Comment (cyprusad) [inline app/models/note.rb:5]: Should provide some prompt in the UI to ensure that comment size is at least 10 characters long

File: config/database.yml [hunk 3]
```diff
@@ -11,15 +11,15 @@ default: &default
...
 # Do not set this db to the same as development or production.
 test:
   <<: *default
-  database: db/test
+  database: screening_test
 
 production:
   <<: *default
-  database: db/production
+  database: screening_production
```
Comment (cyprusad) [inline config/database.yml:25]: Thank you for this! :100:

**Comment by cyprusad** [comment]
@EmilyMB Thank you for doing The Right Thing :tm: 

Just a minor comment about adding UI changes to ensure that comments are at least 10 characters long. Otherwise, :+1:

**Comment by pascallaliberte** [comment]
Noted about the comment length feedback

File: app/models/note.rb (hunk 2 above)
Comment (EmilyMB) [inline app/models/note.rb:5]: My initial thought was to have this in a separate PR, but you're right...this shouldn't go out without the appropriate flash errors; failing silently is the worst.

Comment (pascallaliberte) [inline app/models/note.rb:5]: To get the feedback message right: What are we communicating by requiring a 10-character minimum?

“You wrote a short note. There’s a minimum of 10 characters to make sure you write a long-enough note. Can you add more details?"

Comment (cyprusad) [inline app/models/note.rb:5]: Yeah, something along those lines should be nice. Very empathetic.

**Comment by cyprusad** [comment]
Also, you'll have to rebase with master. Miran an I fixed the broken notes UI last night. :sparkles:

File: app/models/note.rb (hunk 2 above)
Comment (EmilyMB) [inline app/models/note.rb:5]: @pascallaliberte @cyprusad I rebased with @miranmirza's new master added a nice note for users to understand when things didn't save correctly.
screen shot 2015-06-11 at 3 21 42 pm

Comment (cyprusad) [inline app/models/note.rb:5]: :+1: great, thanks!

**Comment by pascallaliberte** [comment]
:+1:

**Pull request merged**